web-client/src/components/MainLayout/index.tsx

```diff
@@ -1,118 +1,143 @@
-                                to={route.to}
-                                color="#1976d2"
-                                className={pathname === route.to ? "active" : ""}>{route.name}</NavLink>
-                        </ListItemButton>
-                    </ListItem>
-                ))}
-            </List>
-        </Box>
-    );
+            window.addEventListener('resize', checkIsMobile);
+            checkIsMobile();
 
-    const container = window !== undefined ? () => window().document.body : undefined;
+            return () => {
+                window.removeEventListener('resize', checkIsMobile);
+            };
+        }
+    }, []);
 
+    const handleChange = (event: SelectChangeEvent) => {
+        setLang(event.target.value as string);
+    };
     return (
-        <Box sx={{ display: 'flex' }}>
-            <CssBaseline />
-            <AppBar component="nav">
-                <Toolbar>
-                    <IconButton
-                        color="inherit"
-                        aria-label="open drawer"
-                        edge="start"
-                        onClick={handleDrawerToggle}
-                        sx={{ mr: 2, display: { sm: 'none' } }}
-                    >
-                        <MenuIcon />
-                    </IconButton>
-                    <Typography
-                        variant="h6"
-                        component="div"
-                        sx={{ flexGrow: 1, display: { sm: 'block' } }}
-                    >
-                        <NavLinkLogo to="/">RC-UA</NavLinkLogo>
-                    </Typography>
-                    <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
-                        <Grid container spacing={2}>
-                            {appRoutesMap.map((route) => (
-                                <Grid item key={route.name}>
-                                    <NavLink to={route.to} className={pathname === route.to ? "active" : ""}>
-                                        {route.name}
-                                    </NavLink>
-                                </Grid>
-                            ))}
-                        </Grid>
+        isMobile ? (
+            <Paper sx={{
+                position: 'fixed', top: 0, bottom: 0, left: 0, right: 0,
+                minHeight: '100vh',
+                display: 'flex',
+                flexDirection: 'column',
+                backgroundColor: "#eaeaed",
+                // padding: '8px'
+            }}>
+                <FlexBox justify="space-between" align="center" backgroundColor="#ffffff" padding="12px 16px 0 16px" height="75px">
+                    <NavLinkLogo to="/"><img src={Logo} alt="Logo" /></NavLinkLogo>
+                    <Box sx={{ width: '100px', height: '40px' }}>
+                        <FormControl fullWidth>
+                            <InputLabel id="demo-simple-select-label">Language</InputLabel>
+                            <Select
```

Comment:
line 59 and following would be great in another file

Reply:
I made this selection a separate component of LangSelect and changed this code block for both mobile and desktop versions, as you wrote. Thanks!